compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1141,51 +1162,61 @@ trait Applications extends Compatibility {
-        * method and one of its arguments has a type mismatch , append
-        * a note to the error message that explains where the required
-        * type comes from. See #19680 and associated test case.
+        *  method and one of its arguments has a type mismatch , append
+        *  a note to the error message that explains where the required
+        *  type comes from. See #19680 and associated test case.
         */
       def maybeAddInsertedApplyNote(failedState: TyperState, fun1: Tree)(using Context): Unit =
         if fun1.symbol.name == nme.apply && fun1.span.isSynthetic then
           fun1 match
-            case Select(qualifier, _) =>
-              def mapMessage(dia: Diagnostic): Diagnostic =
-                dia match
-                  case dia: Diagnostic.Error =>
-                    dia.msg match
-                      case msg: TypeMismatch =>
-                        msg.inTree match
-                          case Some(arg) if tree.args.exists(_.span == arg.span) =>
-                            val noteText =
-                              i"""The required type comes from a parameter of the automatically
-                                  |inserted `apply` method of `${qualifier.tpe}`.""".stripMargin
-                            Diagnostic.Error(msg.appendExplanation("\n\n" + noteText), dia.pos)
-                          case _ => dia
-                      case msg => dia
-                  case dia => dia
-              failedState.reporter.mapBufferedMessages(mapMessage)
-            case _ => ()
-        else ()
+          case Select(qualifier, _) =>
+            def mapMessage(dia: Diagnostic): Diagnostic =
+              dia match
+              case dia: Diagnostic.Error =>
+                dia.msg match
+                case msg: TypeMismatch =>
+                  msg.inTree match
+                  case Some(arg) if tree.args.exists(_.span == arg.span) =>
+                    val noteText =
+                      i"""The required type comes from a parameter of the automatically
+                          |inserted `apply` method of `${qualifier.tpe}`.""".stripMargin
+                    Diagnostic.Error(msg.appendExplanation("\n\n" + noteText), dia.pos)
+                  case _ => dia
+                case msg => dia
+              case dia => dia
+            failedState.reporter.mapBufferedMessages(mapMessage)
+          case _ => ()
+
+      def maybePatchBadParensForImplicit(failedState: TyperState)(using Context): Boolean =
+        def rewrite(): Unit =
+          val replace =
+            if isInsertedApply then ".apply" // x() -> x.apply
+            else "" // f() -> f where fun1.span.end == tree.span.point
+          rewrites.Rewrites.patch(tree.span.withStart(fun1.span.end), replace)
+        var retry = false
+        failedState.reporter.mapBufferedMessages: dia =>
+          dia match
+          case err: Diagnostic.Error =>
```

Comment:
The indentation is off here.

Reply:
I'm an advocate for "match-aligned case" in braceless code. It makes it easier to visually line up indents; otherwise, the last case looks indented too far; and there are too many indents to line up.

It only looks funny because we are used to braces, which visually require an extra indent but also provide the closing brace as a corrective.

However, after offering my standard screed, here I should just remove the match, which might be left over from adding debug println.